dotnet/src/Connectors/Connectors.Memory.AzureAISearch/AzureAISearchVectorStoreCollectionCreateMapping.cs

```diff
@@ -34,8 +34,13 @@ public static SearchableField MapKeyField(VectorStoreRecordKeyProperty keyProper
     /// <exception cref="InvalidOperationException">Throws when the definition is missing required information.</exception>
     public static SimpleField MapDataField(VectorStoreRecordDataProperty dataProperty, string storagePropertyName)
     {
-        if (dataProperty.PropertyType == typeof(string))
+        if (dataProperty.IsFullTextSearchable)
         {
+            if (dataProperty.PropertyType != typeof(string))
+            {
+                throw new InvalidOperationException($"Property {nameof(dataProperty.IsFullTextSearchable)} on {nameof(VectorStoreRecordDataProperty)} '{dataProperty.DataModelPropertyName}' is set to true, but the property type is not a string.");
```

Comment:
Add detail explaining that AzureAISearch requires the property type is a string. If I've written my code against the Volatile data store and everything work s and then it fails when I reconfigure to use Azure AI Search I want to know this is because of restrictions in Azure AI Search.

Reply:
Makes sense, updated.